File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/FileSelection/Filecosen.java
package com.digicoffer.lauditor.CommonFiles.FileSelection;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.digicoffer.lauditor.R;

public class Filecosen {

    private final Activity activity;
    private Dialog dialog;
    private Context context;

    public interface FileChooserCallback {
        void openCamera();
        void openGallery();
    }

    private FileChooserCallback callback;

    public Filecosen(Activity activity, FileChooserCallback callback) {
        this.activity = activity;
        this.context = activity;
        this.callback = callback;
    }

    public void show() {
        dialog = new Dialog(activity);
        dialog.setContentView(R.layout.dialog_file_chosen);
        dialog.setCancelable(true);

        if (dialog.getWindow() != null) {
            dialog.getWindow().setLayout(
                    ViewGroup.LayoutParams.MATCH_PARENT,
                    ViewGroup.LayoutParams.WRAP_CONTENT
            );
            dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            dialog.getWindow().setGravity(Gravity.BOTTOM);
        }

        LinearLayout llCamera = dialog.findViewById(R.id.llCamera);
        LinearLayout llGallery = dialog.findViewById(R.id.llGallery);
        TextView tvCancel = dialog.findViewById(R.id.tvCancel);

        llCamera.setOnClickListener(v -> {
            callback.openCamera();
            dialog.dismiss();
        });

        llGallery.setOnClickListener(v -> {
            callback.openGallery();
            dialog.dismiss();
        });

        tvCancel.setOnClickListener(v -> dialog.dismiss());

        dialog.show();
    }
}

File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/FileSelection/Filecosen.kt
package com.digicoffer.lauditor.CommonFiles.FileSelection

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.Gravity
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.digicoffer.lauditor.R

class Filecosen(private val activity: Activity, private val callback: FileChooserCallback) {

    private var dialog: Dialog? = null
    private val context: Context = activity

    interface FileChooserCallback {
        fun openCamera()
        fun openGallery()
    }

    fun show() {
        val currentDialog = Dialog(activity)
        dialog = currentDialog
        currentDialog.setContentView(R.layout.dialog_file_chosen)
        currentDialog.setCancelable(true)

        currentDialog.window?.let { window ->
            window.setLayout(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
            window.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            window.setGravity(Gravity.BOTTOM)
        }

        val llCamera = currentDialog.findViewById<LinearLayout>(R.id.llCamera)
        val llGallery = currentDialog.findViewById<LinearLayout>(R.id.llGallery)
        val tvCancel = currentDialog.findViewById<TextView>(R.id.tvCancel)

        llCamera.setOnClickListener {
            callback.openCamera()
            currentDialog.dismiss()
        }

        llGallery.setOnClickListener {
            callback.openGallery()
            currentDialog.dismiss()
        }

        tvCancel.setOnClickListener { currentDialog.dismiss() }

        currentDialog.show()
    }
}
